keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/VerificationController.kt

```diff
@@ -33,31 +37,83 @@ class VerificationController(
     @PathVariable("environment") environment: String,
-    @RequestBody status: UpdatedVerificationStatus
+    @RequestBody payload: UpdateVerificationStatusRequest
   ) {
-    require(status.status in listOf(OVERRIDE_PASS, OVERRIDE_FAIL)) {
-      "Only override statuses may be set via this endpoint."
+    if (payload.status !in listOf(OVERRIDE_PASS, OVERRIDE_FAIL)) {
+      throw NotOverrideStatus()
     }
+
     VerificationContext(
       deliveryConfig = deliveryConfigRepository.getByApplication(application),
       environmentName = environment,
-      artifactReference = status.artifactReference,
-      version = status.artifactVersion
+      artifactReference = payload.artifactReference,
+      version = payload.artifactVersion
     ).apply {
       verificationRepository.updateState(
         context = this,
-        verification = verification(status.verificationId),
-        status = status.status,
-        mapOf("overriddenBy" to user, "comment" to status.comment)
+        verification = verification(payload.verificationId),
+        status = payload.status,
+        mapOf("overriddenBy" to user, "comment" to payload.comment)
       )
     }
   }
 
-  data class UpdatedVerificationStatus(
+  data class UpdateVerificationStatusRequest(
     val verificationId: String,
     val artifactReference: String,
     val artifactVersion: String,
     val status: ConstraintStatus,
     val comment: String?
   )
+
+  @ResponseStatus(UNPROCESSABLE_ENTITY)
+  private class NotOverrideStatus :
+    RuntimeException("Only override statuses may be set via this endpoint.")
+
+  @PostMapping(
+    path = ["/{application}/environment/{environment}/verifications/retry"],
+    consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  @PreAuthorize(
+    """@authorizationSupport.hasApplicationPermission('WRITE', 'APPLICATION', #application)
+    and @authorizationSupport.hasServiceAccountAccess('APPLICATION', #application)"""
+  )
+  fun retryVerification(
+    @RequestHeader("X-SPINNAKER-USER") user: String,
+    @PathVariable("application") application: String,
+    @PathVariable("environment") environment: String,
+    @RequestBody payload: RetryVerificationRequest
+  ) {
+    VerificationContext(
+      deliveryConfig = deliveryConfigRepository.getByApplication(application),
+      environmentName = environment,
+      artifactReference = payload.artifactReference,
+      version = payload.artifactVersion
+    ).apply {
+      verificationRepository.getState(
+        context = this,
+        verification = verification(payload.verificationId)
+      )?.apply {
+        if (!status.complete) throw VerificationIncomplete()
+      }
+
+      verificationRepository.updateState(
+        context = this,
+        verification = verification(payload.verificationId),
+        status = NOT_EVALUATED,
+        mapOf("retryRequestedBy" to user)
```

Comment:
Should we include a `retryCount` as well?

Reply:
I was thinking about making the format more complex with an array of objects rather than an array of just the task ids but then it was super hard to get the `json_merge` to work the way I wanted.

Reply:
I meant to the map here. I thought this parameter was just a map of arbitrary metadata?

Reply:
That would assume that it was retried in each case by the same user